src/backend/renderer/element/surface.rs

```diff
@@ -530,14 +547,17 @@ where
         damage: &[Rectangle<i32, Physical>],
         opaque_regions: &[Rectangle<i32, Physical>],
     ) -> Result<(), R::Error> {
-        frame.render_texture_from_to(
-            &self.texture,
-            src,
-            dst,
-            damage,
-            opaque_regions,
-            self.buffer_transform,
-            self.alpha,
-        )
+        match self.texture {
+            WaylandSurfaceTexture::Texture(ref texture) => frame.render_texture_from_to(
+                texture,
+                src,
+                dst,
+                damage,
+                opaque_regions,
+                self.buffer_transform,
+                self.alpha,
+            ),
+            WaylandSurfaceTexture::SolidColor(color) => frame.draw_solid(dst, damage, color),
```

Comment:
The wp-alpha-modifier protocol can be used with single-pixel buffers. We need to accommodate for that here, by blending `self.alpha` with `color`.